include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,372 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/algo/copy.hpp>
+#include <eve/algo/equal.hpp>
+#include <eve/algo/fill.hpp>
+#include <eve/algo/views/convert.hpp>
+
+#include <eve/arch/expected_cardinal.hpp>
+#include <eve/algo/container/detail/soa_storage2.hpp>
+#include <eve/memory/aligned_allocator.hpp>
+#include <eve/product_type.hpp>
+#include <eve/traits.hpp>
+#include <eve/assert.hpp>
+
+#include <memory>
+
+namespace eve::algo
+{
+  //================================================================================================
+  //! @addtogroup memory
+  //! @{
+  //!   @brief SIMD-aware container for product types
+  //!
+  //!   **Required header:** `#include <eve/memory/soa_vector.hpp>`
+  //!
+  //!   soa_vector is a container similar to `std::vector` but optimized to store
+  //!   [product types](eve::product_type) using a Structure Of Array layout compatible with SIMD
+  //!   processing.
```

Comment:
mention one allocation?

Reply:
Will redo the documentation before final merge